### PR TITLE
Added firewall check state to check open ports from a minion to a host

### PR DESCRIPTION
**Pull request by rdutch**
Firewall.check will check a connection to a host from the minion on a predefined port.
Returns True when the port is open, False when the port is closed.

### CODE REVIEW

**Comment by jfindlay** [comment]
@rdutch: https://jenkins.saltstack.com/job/salt-pr-lint-n/7459/violations/file/salt/states/firewall.py/.

**Comment by rdutch** [comment]
Fixed pylint errors

File: salt/states/firewall.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+'''
+State to check firewall configurations
+'''
```
Comment (rallytime) [inline salt/states/firewall.py]: @rdutch Can you add `.. versionadded:: Boron` to the comment block here?

**Comment by rallytime** [comment]
It looks like there are still a couple of pylint errors on this one: https://jenkins.saltstack.com/job/salt-pr-lint-n/7480/violations/file/salt/states/firewall.py/

**Comment by cachedout** [comment]
Go Go Jenkins!

**Comment by cachedout** [comment]
This still has lint errors but I will fix them in a separate PR.

**Pull request merged**